### DISCUSSION TITLE
How can I stop Vercel preview deployments from appearing on pull requests?

### DISCUSSION THREAD

**Discussion by mbokinala** [vercel/vercel#6434] [Help] [answered]
Vercel automatically deploys code pushed to my master branch to production, and I would like to keep this functionality. However, Vercel also creates preview deployments for each push, and when I create a pull request, this deployment shows up under 'Checks.' Is there a way to stop this from happening? I don't have an issue with Vercel creating a preview deployment for each push I make, but I would like for Vercel's deployment not to appear under the 'Checks' section of each pull request.

**Comment by barelyhuman** [comment, marked as answer]
While I understand you don't want the preview to show on the the PR's , you can kind of disable the build step for all other branches which a shell command that vercel currently executes. 

You can find this in Project> Example Project > Settings > Git > Ignored Build step, this will not create new builds and thus never create a preview or update the PR based on the command you put in there. 

Which can be a single line command to check for branch | push | etc. 
This is not the perfect solution, but a hack for now. 

You can read more about it here.
https://vercel.com/docs/platform/projects#ignored-build-step

**Reply by mbokinala** [reply]
Will this still deploy when I merge into `master`?

**Reply by barelyhuman** [reply]
if you change the ignore build step to something like 
`if [[ "$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')" != "master" ]] then; exit 0; fi`
or the alternative
`if [[ "$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')" == "master" ]] then; exit 1; fi`

where the condition returns 1 when it needs a new build, so that is everytime the branch is master it'll run the build, though this is theoretical and you could test it and if it works then you are good to go. the above will signal a new build on the platform

**Reply by mbokinala** [reply]
Thank you very much, this worked. This page has more details for anyone else looking to do something similar: https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel

**Reply by 3kh0** [reply]
thanks!!!!

**Reply by dickyindra** [reply]
I get an error during build: `/bin/sh: -c: line 0: syntax error near unexpected token 'then'`

![image](https://user-images.githubusercontent.com/13186704/187131471-ac2e8ea5-25c5-4919-b046-47d572220b6e.png)

**Reply by HappySean2845** [reply]
if [[ "$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')" != "next" ]] ; then exit 0; fi

**Reply by kvedantmahajan** [reply]
We have "require status check before merging flag" for all our repos. Using ignore build step, makes this status check false. Therefore, I am not able to merge my pull request. Any suggestions on the same?

**Reply by ADTC** [reply]
I think this is too complex for something that could easily be some simple configurations like these:

> Build pushed branches:
> (o) Always.
> (o) Only when a pull request is opened.
>
> Only build branches matching regex: [_____________]
> 
> Ignore branches matching regex: [_____________]

The scripted control can be for more advanced scenarios. **I don't even know how to script for the first one (where a build happens only when a PR is opened).**

**Reply by addlistener** [reply]
I checked that git branch returns master on vercel. I don't know why...

**Comment by priyankarkoley** [comment]
> Vercel automatically deploys code pushed to my master branch to production, and I would like to keep this functionality. However, Vercel also creates preview deployments for each push, and when I create a pull request, this deployment shows up under 'Checks.' Is there a way to stop this from happening? I don't have an issue with Vercel creating a preview deployment for each push I make, but I would like for Vercel's deployment not to appear under the 'Checks' section of each pull request.


There is a direct feature in case of vercel, that is quite unpopular. Choose a project from the dashboard and go to its Settings tab. Under the **GIT** section of the Settings tab, there is a dropdown for **"Ignored Build Step"**, where you can select `Only build production`.

For me, this worked fine. And, I would highly recommend vercel "**hobby**" tier users to make a seperate branch from main(maybe 'dev'), and push your updates in that branch. And, merge it to **main** periodically when you need to propagate the updates to production. In that way you'll save some of your limited(probably in millions, but limited) build time.

![Screenshot 2024-07-15 112835](https://github.com/user-attachments/assets/37cf8862-0cd6-4387-a72a-d0237ed2071d)
